cmd/rollkit/commands/run_node.go

```diff
@@ -213,40 +214,91 @@ func NewRunNodeCmd() *cobra.Command {
 				return fmt.Errorf("failed to create new rollkit node: %w", err)
 			}
 
-			// Start the node
-			if err := rollnode.Start(ctx); err != nil {
+			// Create error channel and signal channel
+			errCh := make(chan error, 1)
+			shutdownCh := make(chan struct{})
+
+			// Start the node in a goroutine
+			go func() {
+				err := rollnode.Run(ctx)
+				select {
+				case errCh <- err:
+				default:
+					logger.Error("Error channel full", "error", err)
+				}
+			}()
+
+			// Wait a moment to check for immediate startup errors
+			time.Sleep(100 * time.Millisecond)
+
+			// Check if the node stopped immediately
+			select {
+			case err := <-errCh:
 				return fmt.Errorf("failed to start node: %w", err)
+			default:
+				// This is expected - node is running
+				logger.Info("Started node")
 			}
 
-			// TODO: Do rollkit nodes not have information about them? CometBFT has node.switch.NodeInfo()
-			logger.Info("Started node")
-
 			// Stop upon receiving SIGTERM or CTRL-C.
-			cometos.TrapSignal(logger, func() {
-				if rollnode.IsRunning() {
-					if err := rollnode.Stop(ctx); err != nil {
-						logger.Error("unable to stop the node", "error", err)
-					}
-				}
-			})
+			go func() {
+				cometos.TrapSignal(logger, func() {
+					logger.Info("Received shutdown signal")
+					cancel() // Cancel context to stop the node
+					close(shutdownCh)
+				})
+			}()
 
 			// Check if we are running in CI mode
 			inCI, err := cmd.Flags().GetBool("ci")
 			if err != nil {
 				return err
 			}
+
 			if !inCI {
-				// Block forever to force user to stop node
-				select {}
+				// Block until either the node exits with an error or a shutdown signal is received
+				select {
+				case err := <-errCh:
+					return fmt.Errorf("node exited with error: %w", err)
+				case <-shutdownCh:
+					// Wait for the node to clean up
+					select {
+					case <-time.After(5 * time.Second):
+						logger.Info("Node shutdown timed out")
+					case err := <-errCh:
+						if err != nil && !errors.Is(err, context.Canceled) {
+							logger.Error("Error during shutdown", "error", err)
+						}
+					}
+					return nil
+				}
 			}
 
-			// CI mode. Wait for 5s and then verify the node is running before calling stop node.
+			// CI mode. Wait for 5s and then verify the node is running before cancelling context
 			time.Sleep(5 * time.Second)
```

Comment:
Maybe we should print something before sleep to calm down impatient users?

Reply:
this is only when using the --ci mode. so users shouldnt even see the wait time.

Reply:
this is only when using the --ci mode. so users shouldnt even see the wait time.